src/profile-logic/profile-data.js

```diff
@@ -2085,6 +2086,67 @@ export function filterToDeallocations(
   return newNativeAllocations;
 }
 
+/**
+ * This function filters to only negative memory size values in the native allocations.
+ * It rewrites the stacks to point back to the stack of the allocation.
+ */
+export function filterToDeallocationsMemory(
+  nativeAllocations: BalancedNativeAllocationsTable
+): NativeAllocationsTable {
+  // A-----D------A-------D
+  type Address = number;
+  type IndexIntoAllocations = number;
+  const memoryAddressToAllocation: Map<
```

Comment:
That logic looks pretty similar to retained allocations logic. I wonder if we could merge those. But I'm okay to land it like this since they are not identical after all.